clang/include/clang/Basic/Attr.td

```diff
@@ -3823,3 +3823,10 @@ def EnforceTCBLeaf : InheritableAttr {
   let Documentation = [EnforceTCBLeafDocs];
   bit InheritEvenIfAlreadyPresent = 1;
 }
+
+def YkNoTrace : InheritableAttr {
+  let Spellings = [GCC<"yk_no_trace">, Declspec<"yk_no_trace">];
```

Comment:
I notice here it's `yk_no_trace` but in `CodeGenModule.cpp` it's `yk-no-trace`?

Reply:
Backing out the compiler change breaks the tests, so I'm certain it's working. You should check the tests in the companion PR to be confident that I'm correctly checking for calls to the outlined function.

Prodding around in the clang sources, these "spellings" appear to be used for documentation and error reporting (but it's hard to fully understand, as the file is the input for a custom table generator).

The unique ID is correct in all places, so that's probably why it's working.

Regardless, I'm now going to rename everything to `YkOutline`/`yk-outline`.

Reply:
[a545abc](https://github.com/ykjit/ykllvm/pull/28/commits/a545abca38b498e8da23824198f17976d0854878) does the renaming.

I think I was confused yesterday. No existing clang attributes use hyphens, so nor should we. You can verify my claim [here](https://clang.llvm.org/docs/AttributeReference.html).

I also:
 - pushed a commit to the companion.
 - removed the annotation from a test to confirm that it fails the test. It does.
 - manually eyeballed the IR when removing the attribute to check that it did indeed inline.
 - Put the annotation back, eyeballed again. All is well.


Outlined:
```
  ; main() tests/no_trace_annotation2.c:64:5                                                          
  call void @call_me() #0, !dbg !30                                                                   
  ; main() tests/no_trace_annotation2.c:65:6                                                          
  %28 = load i32, i32* %11, align 4, !dbg !31, !tbaa !32 
```

vs. inlined:

```                           
  ; call_me() tests/no_trace_annotation2.c:48:11                                                      
  %40 = load %struct._IO_FILE*, %struct._IO_FILE** @stderr, align 8, !dbg !13                         
  ; call_me() tests/no_trace_annotation2.c:48:3                                                       
  %41 = call i32 (%struct._IO_FILE*, i8*, ...) @fprintf(%struct._IO_FILE* %40, i8* getelementptr inbounds ([17 x i8], [17 x i8]* @.str.1, i64 0, i64 0)), !dbg !17
  ; call_me2() tests/no_trace_annotation2.c:45:31                                                     
  %42 = load %struct._IO_FILE*, %struct._IO_FILE** @stderr, align 8, !dbg !18                         
  ; call_me2() tests/no_trace_annotation2.c:45:23                                                     
  %43 = call i32 (%struct._IO_FILE*, i8*, ...) @fprintf(%struct._IO_FILE* %42, i8* getelementptr inbounds ([10 x i8], [10 x i8]* @.str, i64 0, i64 0)), !dbg !20
  ; main() tests/no_trace_annotation2.c:63:6                                                          
  %44 = load i32, i32* %19, align 4, !dbg !21  
```

You can see in the latter from the debug comments, that the IR statements came from the inlined callees.